common/browser_context.go

```diff
@@ -70,39 +74,82 @@ func (b *BrowserContext) AddCookies(cookies goja.Value) *goja.Promise {
 }
 
 // AddInitScript adds a script that will be initialized on all new pages.
-func (b *BrowserContext) AddInitScript(script goja.Value, arg goja.Value) {
+func (b *BrowserContext) AddInitScript(script goja.Value, arg goja.Value) error {
 	b.logger.Debugf("BrowserContext:AddInitScript", "bctxid:%v", b.id)
 
-	rt := b.vu.Runtime()
+	if !gojaValueExists(script) {
+		return errors.New("parsing init script: invalid script")
+	}
 
-	source := ""
-	if script != nil && !goja.IsUndefined(script) && !goja.IsNull(script) {
-		switch script.ExportType() {
-		case reflect.TypeOf(string("")):
-			source = script.String()
-		case reflect.TypeOf(goja.Object{}):
-			opts := script.ToObject(rt)
-			for _, k := range opts.Keys() {
-				switch k {
-				case "content":
-					source = opts.Get(k).String()
-				}
-			}
-		default:
-			_, isCallable := goja.AssertFunction(script)
-			if !isCallable {
-				source = fmt.Sprintf("(%s);", script.ToString().String())
-			} else {
-				source = fmt.Sprintf("(%s)(...args);", script.ToString().String())
+	_, isCallable := goja.AssertFunction(script)
+	hasArgs := gojaValueExists(arg)
+
+	var err error
+	var source string
+	//nolint:exhaustive
+	switch script.ExportType().Kind() {
```

Comment:
This could go to a refactoring commit first.

`script.ExportType()` -> `...Kind()` and the cases.

Also, the `parseObjectInitScript` function, could go to a refactoring one.